### PR TITLE
Rework unit test to common test

### PR DESCRIPTION
**Pull request by platinumthinker**

### CODE REVIEW

**Comment by surik** [comment]
Can you also remove eunit task from `travis.yml`?

File: include/eradius_lib.hrl [hunk 1]
```diff
@@ -1,5 +1,13 @@
 -define(BYTE, integer-unit:8).    % Nice syntactic sugar...
 
+-define(DEFAULT_TYPE, realm).
```
Comment (surik) [inline include/eradius_lib.hrl]: I'm not sure that it is a good place for proxy related defines.

File: src/eradius_lib.erl [hunk 2]
```diff
@@ -8,7 +8,8 @@
 
 -ifdef(TEST).
 -include_lib("eunit/include/eunit.hrl").
```
Comment (RoadRunnr) [inline src/eradius_lib.erl]: the eunit reference should be remove

File: test/eradius_config_SUITE.erl [hunk 3]
```diff
@@ -10,6 +10,7 @@
 -module(eradius_config_SUITE).
 -compile(export_all).
 -include("../include/eradius_lib.hrl").
+-include_lib("eunit/include/eunit.hrl").
```
Comment (RoadRunnr) [inline test/eradius_config_SUITE.erl]: no eunit references in common test please. use the ?match and ?equal macros that we use elsewhere (e.g. https://github.com/travelping/gtplib/blob/master/test/gtp_SUITE.erl#L22)

File: test/eradius_lib_SUITE.erl [hunk 4]
```diff
@@ -56,3 +89,79 @@ ipv6prefix(_Config) ->
...
+-define(USER, "test").
+-define(SECRET, <<"secret">>).
+-define(PLAIN_TEXT, "secret").
+-define(PLAIN_TEXT_PADDED, <<"secret",0,0,0,0,0,0,0,0,0,0>>).
+-define(CIPHER_TEXT, <<171,213,166,95,152,126,124,120,86,10,78,216,190,216,26,87,55,15>>).
+-define(ENC_PASSWORD, 186,128,194,207,68,25,190,19,23,226,48,206,244,143,56,238).
+-define(PDU, #radius_request{ reqid = 1, secret = ?SECRET, authenticator = ?REQUEST_AUTHENTICATOR }).
+
+selt_encrypt_test(_) ->
+    ?CIPHER_TEXT = eradius_lib:salt_encrypt(?SALT, ?SECRET, ?REQUEST_AUTHENTICATOR, << ?PLAIN_TEXT >>).
```
Comment (RoadRunnr) [inline test/eradius_lib_SUITE.erl]: ?equal(...)

File: test/eradius_proxy_SUITE.erl [hunk 5]
```diff
@@ -0,0 +1,104 @@
+%    __                        __      _
+%   / /__________ __   _____  / /___  (_)___  ____ _
+%  / __/ ___/ __ `/ | / / _ \/ / __ \/ / __ \/ __ `/
+% / /_/ /  / /_/ /| |/ /  __/ / /_/ / / / / / /_/ /
+% \__/_/   \__,_/ |___/\___/_/ .___/_/_/ /_/\__, /
+%                           /_/            /____/
+%
+% Copyright (c) Travelping GmbH <info@travelping.com>
+%
+
```
Comment (RoadRunnr) [inline test/eradius_proxy_SUITE.erl]: add LICENSE text, copy from eradius_lib_SUITE.erl

**Comment by RoadRunnr** [comment]
also, please rework the commit, squash where sensible. From what it see at the moment it should be two commit, one the fix the whitespaces and a second for eunit/ct conversion.

And if you have time, add a third cset with additional tests for strip_test:
```
?equal("user/user2@example",  eradius_proxy:strip("roaming/user/user2@example", prefix, true, "/")),
?equal("user/user2@example",  eradius_proxy:strip("user/user2@example@roaming", realm, true, "@")),
```

File: src/eradius_config.erl [hunk 6]
```diff
@@ -343,35 +343,18 @@ ok_error_helper({error, _Error}, ErrorMsg) when is_list(ErrorMsg) -> ErrorMsg;
...
     generate_ip_list(IP, catch list_to_integer(Mask));
-generate_ip_list({A, B, C, D}, Mask) when Mask >=0, Mask =< 32 -> 
+generate_ip_list({A, B, C, D}, Mask) when Mask >=0, Mask =< 32 ->
     <<Address:32/integer>> = <<A, B, C, D>>,
     Wildcard = 16#ffffffff bsr Mask,
     <<Netmask:32/unsigned-integer>> = << (bnot Wildcard):32 >>,
     generate_ip(Address band Netmask, Address bor Wildcard);
 generate_ip_list(_, Mask) -> ?invalid("invalid mask ~p", [Mask]).
 
-generate_ip(E, E) ->
```
Comment (RoadRunnr) [inline src/eradius_config.erl]: these white space changes should go into a separate commit

**Comment by platinumthinker** [comment]
I separate pr on two commits

**Comment by surik** [comment]
I still think we shouldn't place proxy defaults into `include/eradius_lib.hrl`

**Comment by platinumthinker** [comment]
@surik, i done

File: src/eradius_proxy.erl [hunk 7]
```diff
@@ -3,12 +3,18 @@
...
+         get_key/4, strip/4]).
+-endif.
+
 -include("eradius_lib.hrl").
 -include("dictionary.hrl").
 
 -define(DEFAULT_TYPE, realm).
 -define(DEFAULT_STRIP, false).
 -define(DEFAULT_SEPARATOR, "@").
+
```
Comment (RoadRunnr) [inline src/eradius_proxy.erl]: is there a reason to have the default type, strip and separator as defines? I can see only one place where they are used.

Comment (RoadRunnr) [inline src/eradius_proxy.erl]: Is there a point in keep the defaults for type, strip and separator as defines? There where used in the eunit test, but there is only one place remaining that is using them.

Comment (RoadRunnr) [reply src/eradius_proxy.erl]: should have read further, actually the DEFAULT_OPTIONS define is obsolete. The places where it is used could be replaced with ```[]```

Comment (surik) [reply src/eradius_proxy.erl]: It seems that we cant put `[]` into `validate_options/1`. It needs all default options to validate. In the scope of this PR I would save the defines.

**Comment by RoadRunnr** [comment]
@surik commented on this pull request. 

> In [ https://github.com/travelping/eradius/pull/110#discussion_r141569882 |
> src/eradius_proxy.erl ] :
> >  -include("eradius_lib.hrl").
>  -include("dictionary.hrl").

>  -define(DEFAULT_TYPE, realm).
>  -define(DEFAULT_STRIP, false).
>  -define(DEFAULT_SEPARATOR, "@").
> +

> It seems that we cant put [] into validate_options/1 . It needs all default
> options to validate. In the scope of this PR I would save the defines.

The whole options handling is a complete mess anyway. Lets do the cleanup another time.
